lib/pusher-platform/sdk_info.rb

```diff
@@ -0,0 +1,26 @@
+require_relative './error'
+
+module PusherPlatform
+  class SDKInfo
+    attr_reader :product_name, :version, :language, :platform, :headers
+
+    def initialize(options)
+      raise Error.new('No product_name provided to SDKInfo') if options[:product_name].nil?
+      raise Error.new('No version provided to SDKInfo') if options[:version].nil?
+
+      @product_name = options[:product_name]
+      @version = options[:version]
+
+      @platform = options[:platform] || 'server'
+      @language = 'ruby'
+
+      @headers = {
+        "X-SDK-Product" => @product_name,
```

Comment:
I'm far from an expert on the SDKs or Ruby. These headers seem decided now, but I would have expected to see them start `X-Pusher-SDK`.

Otherwise, LGTM.

Reply:
Probably would have been wise. AFAIK nobody is actually using them right now, so if it makes sufficient sense to change them, we should do it now. WDYT?

Reply:
We'd have to update all SDKs with new headers, I don't know if you'll find volunteers for that :P

Reply:
Thing is, it's only Chatkit, so if we ever want to do it, now's the time. If we're in agreement that it would be better to have `X-Pusher-SDK-Product`, `X-Pusher-SDK-Language`, `X-Pusher-SDK-Platform` and `X-Pusher-SDK-Version` then I can do that today

Reply:
Yes, updating everything now doesn't sound like fun.

However, the `X-` prefix was deprecated 6 years ago in [rfc6648](https://tools.ietf.org/html/rfc6648), but will, of course, likely continue to do as expected until the heat death of the internet.

If we did want to change the form, we could consider not using headers, or using fewer headers to express the intent. Isn't there an implication on `Access-Control-Expose-Headers`?

If we're going to make any large changes, I suggest we make it part of elements protocol.

Reply:
> Isn't there an implication on Access-Control-Expose-Headers?

I don't think so. Isn't that about clients being able to read response headers? These headers are request headers that the client is sending.

> If we're going to make any large changes, I suggest we make it part of elements protocol.

Not sure what counts as a large change.

Chatkit needs whatever we go with (whether there's a change or not) to be working and stable in the very near future. 

If you have specific ideas then now is the time, otherwise it may well end up never happening.

As it stands, I'm happy to drop the `X-` prefix and move to using:

* `Pusher-SDK-Product`
* `Pusher-SDK-Language`
* `Pusher-SDK-Platform` 
* `Pusher-SDK-Version`